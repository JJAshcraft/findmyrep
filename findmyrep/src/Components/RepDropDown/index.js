import React, { Component, Fragment } from "react";
import { withStyles } from "@material-ui/core/styles";
import FormControl from "@material-ui/core/FormControl";
import NativeSelect from "@material-ui/core/NativeSelect";

const styles = theme => ({
  root: {
    display: "flex",
    flexWrap: "wrap"
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 240
  },
  selectEmpty: {
    marginTop: 0
  }
});

class RepDropDown extends Component {
  state = {
    title: "",
    name: "hai",
    labelWidth: 0
  };

  handleChange = name => event => {
    this.props.disableForm(event.target.value);
    this.setState({ [name]: event.target.value });
  };

  render(props) {
    const { classes } = this.props;
    return (
      <Fragment>
        <FormControl className={classes.formControl}>
          <NativeSelect
            value={this.state.age}
            onChange={this.handleChange("title")}
            className={classes.selectEmpty}
          >
            <option value="" />
            <option value={"Representatives"}>Representatives</option>
            <option value={"Senators"}>Senators</option>
          </NativeSelect>
        </FormControl>
      </Fragment>
    );
  }
}

export default withStyles(styles)(RepDropDown);
