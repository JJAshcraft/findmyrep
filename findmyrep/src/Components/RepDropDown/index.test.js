import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import RepDropDown from "./index.js";

test("RepDropDown component renders with proper styling", () => {
  const tree = renderer.create(<RepDropDown />).toJSON();
  expect(tree).toMatchSnapshot();
});
