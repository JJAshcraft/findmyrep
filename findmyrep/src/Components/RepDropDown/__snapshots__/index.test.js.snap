// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepDropDown component renders with proper styling 1`] = `
<div
  className="MuiFormControl-root-4 RepDropDown-formControl-2"
>
  <div
    className="MuiInputBase-root-28 MuiInput-root-15 MuiInput-underline-19 MuiInputBase-formControl-29 MuiInput-formControl-16 RepDropDown-selectEmpty-3"
    onClick={[Function]}
  >
    <div
      className="MuiNativeSelect-root-8"
    >
      <select
        aria-invalid={false}
        className="MuiNativeSelect-select-9 MuiInputBase-input-38 MuiInput-input-23"
        disabled={false}
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
      >
        <option
          value=""
        />
        <option
          value="Representatives"
        >
          Representatives
        </option>
        <option
          value="Senators"
        >
          Senators
        </option>
      </select>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-45 MuiNativeSelect-icon-14"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
